assets/extensions/store.js

```diff
@@ -0,0 +1,284 @@
+/**
+ * External dependencies
+ */
+import { keyBy, merge } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { registerStore, select, dispatch } from '@wordpress/data';
+import { controls, apiFetch } from '@wordpress/data-controls';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { createReducerFromActionMap } from '../shared/data/store-helpers';
+
+/**
+ * Extension statuses.
+ */
+const STATUS = {
+	IN_PROGRESS: 'in-progress',
+	IN_QUEUE: 'in-queue',
+};
+
+/**
+ * Store name.
+ */
+export const EXTENSIONS_STORE = 'sensei/extensions';
+
+/**
+ * Default store state.
+ */
+const DEFAULT_STATE = {
+	// Extensions list to be mapped using the entities.
+	extensionSlugs: [],
+	entities: { extensions: {} },
+	layout: [],
+	error: null,
+};
+
+/**
+ * Checks whether status is a loading status.
+ *
+ * @param {string} status Status to check.
+ *
+ * @return {string} Whether is a loading status.
+ */
+export const isLoadingStatus = ( status ) =>
+	Object.values( STATUS ).includes( status );
```

Comment:
Seems safer to be explicit here instead of including all the statuses. 

```suggestion
	[ STATUS.IN_PROGRESS, STATUS.IN_QUEUE ].includes( status );
```